### ISSUE TITLE
findDOMNode will be deprecated

### ISSUE THREAD

**Issue by dsempel** [joshwnj/react-visibility-sensor#141]
Starting from React 16.6 it is deprecated in StrictMode and it will be deprecated from React in the future. [Here](https://reactjs.org/docs/strict-mode.html#warning-about-deprecated-finddomnode-usage) is some more information about this.

The docs state the following:
> Normally you don’t need this because you can attach a ref directly to a DOM node.

However, reading through the source code I see that VisibilitySensor is not rendering any DOM nodes itself, so I'm not sure what would be the best way to attach the ref.

**Comment by joshwnj** [comment]
Thanks for the heads up @dsempel ! Will have a think about this one

**Comment by donperi** [comment]
@joshwnj Hope the above PR works for you.

**Comment by tanyaburlakova** [comment]
@joshwnj hey!

Can you please help me, I updated my app with webpack 4 and babel 7 and react-visibility-sensor is stooped working.  Do you have any idea why?

Thanks!

**Comment by dancrew32** [comment]
Hey gang, any chance this will hit master? Thanks!

**Comment by RoninSTi** [comment]
Any updates on #142 being merged?

**Comment by alizahir23** [comment]
hey can #142 please be merged into master if its working

**Comment by nigellima** [comment]
Hello. Any changes this PR will be merged  #142? The warning is still present once the component is used

**Comment by nick-0101** [comment]
Any reasons why #142 has not been merged yet?

Issue #179 is right. This project is dead, no use waiting on something that wont be merged.

**Comment by johnbonds** [comment]
+1 on merging this please!

**Comment by nigellima** [comment]
+1 on merging this, please.

**Comment by ComradeLV** [comment]
+1 on merging, please!

**Comment by Vir-al** [comment]
+1 on merging #142, please!

**Comment by sajadko** [comment]
+1 on merging, please!

**Comment by ZengLawrence** [comment]
+1 on merging #142, please!

**Comment by slavagoreev** [comment]
Seems that this issue is stalling for ever

Temporal solution for you guys:

```tsx // index.ts
if (appSettings.isDevelopment) {
  const backup = console.error;

  console.error = function filterWarnings(msg) {
    const supressedWarnings = ['Learn more about using refs'];

    if (!supressedWarnings.some((entry) => msg.includes(entry))) {
      backup.apply(console, arguments);
    }
  };
}
```

**Comment by Seybo** [comment]
+1 on merging, please!

**Comment by myadzel** [comment]
+1 on merging https://github.com/joshwnj/react-visibility-sensor/pull/142, please!

**Comment by ThomasDEVio** [comment]
Merge it!!!

**Comment by ahmadalfy** [comment]
2023, requesting a merge

**Comment by nathggns** [comment]
Has anyone forked this project? It's not maintained anymore, so seems sensible to consider forking it to solve this particular problem.

**Comment by ahmadalfy** [comment]
> Has anyone forked this project? It's not maintained anymore, so seems sensible to consider forking it to solve this particular problem.

I replaced it with [react-waypoint](https://www.npmjs.com/package/react-waypoint). It's almost a drop-in replacement
